src/models/NavItemPage.php

```diff
@@ -298,13 +298,7 @@ private function renderPlaceholderRecursive($navItemPageId, $placeholderVar, $pr
                     $prevIsEqual = array_key_exists($prev, $placeholders) && $placeholder['block_id'] == $placeholders[$prev]['block_id'];
                     $blockObject->setEnvOption('isPrevEqual', $prevIsEqual);
                     $blockObject->setEnvOption('isNextEqual', array_key_exists($next, $placeholders) && $placeholder['block_id'] == $placeholders[$next]['block_id']);
-
-                    if (!$prevIsEqual) {
-                        $equalIndex = 1;
-                    } else {
-                        $equalIndex++;
-                    }
-                    $blockObject->setEnvOption('equalIndex', $equalIndex);
+                    $blockObject->setEnvOption('equalIndex', (!$prevIsEqual) ? $equalIndex = 1 : ++$equalIndex);
```

Comment:
@hbugdoll so this means, all blocks which have been using equalIndex to check whether they are the same block in the list of blocks will result in a different value. This is somewhat considered as breaking change, but since its a bugfix thats ok to me, but we need to add an info to the upgrade.md document. Would you please be so kind to add a short not in the upgrade.md doc what is expected after this change and what was the behahvior before? thank you so much!

Reply:
from the docs (https://luya.io/guide/cms/blocks.html#env-context-information):

> equalIndex: Get the current index/position of this element within the list of same elements.

Reply:
@nadar In my opinion it's not a BC. Please not to be confused with `prev_id` of `NavItemPageBlockItem`.

The calculation rule of `'equalIndex'` hasn't changed! I only harmonized the code in both frontend and admin context.

Here is an example (BlockItem2, BlockItem3 and BlockItem4 are equal):
| Block Item  | `'block_id'` | `'index'` | `'isPrevEqual'`| `'isNextEqual'`| `'equalIndex'` |
| ------------- | :----------: | :-----------: | :-----------: | :-----------: | :--------: |
| BlockItem1  | `1` | `1` | `false` | `false`  | `1`|
| BlockItem2  | `2` | `2` | `false` | `true` | `1`|
| BlockItem3  | `2` | `3` | `true` | `true` | `2`|
| BlockItem4  | `2` | `4` | `true` | `false` | `3`|
| BlockItem5  | `3` | `5` | `false` | `false` | `1`|